packages/ord-connect/src/components/SelectWalletModal/index.tsx

```diff
@@ -171,6 +174,57 @@ export function SelectWalletModal({
     updatePublicKey,
     updateWallet,
   ]);
+  const onConnectLeatherWallet = useCallback(async () => {
+    try {
+      setErrorMessage("");
+      const leather = await getLeatherAddresses(network);
+      if (!leather || leather.length < 1) {
+        disconnectWallet();
+        throw new Error("Leather via Ordit returned no addresses.");
+      }
+
+      const segwit = leather.find(
+        (walletAddress) => walletAddress.format === "segwit",
+      );
+      const taproot = leather.find(
+        (walletAddress) => walletAddress.format === "taproot",
+      );
+
+      if (!segwit || !taproot) {
+        throw new Error(
+          "Leather via Ordit did not return Segwit or Taproot addresses.",
+        );
+      }
```

Comment:
also let's separate the error so that we know which is missing.

Reply:
updated